app-starters-common/app-starters-file-common/src/main/java/org/springframework/cloud/stream/app/file/FileConsumerProperties.java

```diff
@@ -33,49 +33,49 @@
 	 * 'contents' - the contents as bytes.
 	 * Default is 'contents'
 	 */
-	private FileReadingMode fileReadingmode = FileReadingMode.contents;
+	private FileReadingMode mode = FileReadingMode.contents;
 
 	/**
-	 * 	Set to true to emit start of file/end of file marker messages before/after the data. 
+	 * 	Set to true to emit start of file/end of file marker messages before/after the data.
 	 * 	Only valid with FileReadingMode 'lines'.
 	 */
 	private Boolean withMarkers = null;
 
+	/**
+	 * When {@code fileMarkers == true}, specify if they should be produced
+	 * as {@link org.springframework.integration.file.splitter.FileSplitter.FileMarker}
+	 * objects or JSON.
+	 */
+	private boolean markersJson = false;
```

Comment:
M-m-m. I left it as is for backward compatibility.
Although that `true` has bumped my head as well.
Since we don't have `RELEASE` for the stuff yet, I'm sure that will be safe to make it `true` by default.
But at the same time we should be sure that we have Jackson as a dependency.
Let me check what's going on there.